src/main/java/org/mockito/listeners/VerificationListener.java

```diff
@@ -0,0 +1,18 @@
+package org.mockito.listeners;
+
+import org.mockito.verification.VerificationEvent;
+
+/**
+ * This listener can be notified of verify invocations on a mock.
+ * <p>
+ * For this to happen, it must be registered using {@link org.mockito.internal.progress.MockingProgress#addListener(MockitoListener)}.
+ */
+public interface VerificationListener extends MockitoListener {
+    /**
+     * Called after a verification happened.
+     * This includes verifications that fail due to the verification mode failing.
+     *
+     * @param verificationEvent the event that occurred.
+     */
+    void onVerification(VerificationEvent verificationEvent);
```

Comment:
I'm picky but I don't want the term `Event` there, I'd prefer something like `InvocationVerification`